File: src/main/java/io/github/malczuuu/iemu/lwm2m/FirmwareUpdateDeliveryMethod.java
package io.github.malczuuu.iemu.lwm2m;

public enum FirmwareUpdateDeliveryMethod {
  PULL_ONLY(0),
  PUSH_ONLY(1),
  BOTH(2);

  public static FirmwareUpdateDeliveryMethod initial() {
    return BOTH;
  }

  public static FirmwareUpdateDeliveryMethod fromValue(Integer value) {
    if (value == null) {
      return PULL_ONLY;
    }
    return fromValue(value.intValue());
  }

  public static FirmwareUpdateDeliveryMethod fromValue(int value) {
    for (FirmwareUpdateDeliveryMethod mode : values()) {
      if (mode.getValue() == value) {
        return mode;
      }
    }
    return PULL_ONLY;
  }

  private final int value;

  FirmwareUpdateDeliveryMethod(int value) {
    this.value = value;
  }

  public int getValue() {
    return value;
  }
}

File: src/main/kotlin/io/github/malczuuu/iemu/infra/lwm2m/FirmwareUpdateDeliveryMethod.kt
package io.github.malczuuu.iemu.infra.lwm2m

enum class FirmwareUpdateDeliveryMethod(val value: Int) {
  PULL_ONLY(0),
  PUSH_ONLY(1),
  BOTH(2);

  companion object {
    fun initial(): FirmwareUpdateDeliveryMethod = BOTH

    fun fromValue(value: Int?): FirmwareUpdateDeliveryMethod =
        entries.firstOrNull { it.value == value } ?: PULL_ONLY
  }
}
